components/ClassCard.tsx

```diff
@@ -14,25 +15,25 @@ const ClassCard: React.FC<ClassCardProps> = ({
   name,
   minLevel,
   maxLevel,
-  recurrence,
+  rrstring,
   timeStart,
   timeEnd,
 }) => {
-  const weekday: string[] = [
-    "Sunday",
-    "Monday",
-    "Tuesday",
-    "Wednesday",
-    "Thursday",
-    "Friday",
-    "Saturday",
-  ];
+  const rule = RRule.fromString(rrstring);
+  const tempRule = rule.toText();
+  //geting all days of week from the rrule object to output
+  const dates = tempRule.substring(tempRule.indexOf("on") + 3, tempRule.indexOf("until"));
+  //this takes in teh start and end times, converts them to ISO format, then outputs the hour the class starts and ends
   const convertTime = (startTime: string, endTime: string) => {
-    const startTimeStr: string = moment(moment().format(startTime), "HH:mm").format("h");
-    const endTimeStr: string = moment(moment().format(endTime), "HH:mm").format("h A");
-    return startTimeStr + "-" + endTimeStr;
+    const { DateTime } = require("luxon");
+    const startTime_ISO = DateTime.fromISO(startTime);
+    const endTime_ISO = DateTime.fromISO(endTime);
```

Comment:
this unfortunately doesn't do what wre need it to do. We need to convert from the given ISO string, to the local time. So first, we need do `DateTime.fromISO(startTime).toLocal().toFormat('h')` to get the correctly formatted start time. You can construct `finalTime` with that